plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/ui/DBPPlatformUI.java

```diff
@@ -132,6 +132,15 @@ UserChoiceResponse showUserChoice(
 
     void executeWithProgress(@NotNull DBRRunnableWithProgress runnable) throws InvocationTargetException, InterruptedException;
 
+    /**
+     * Execute runnable task synchronously while displaying job indicator if needed
+     */
+    @NotNull
+    <RESULT> RESULT executeWithProgressBlocking(
+        @NotNull String operationDescription,
+        @NotNull RunnableWithResult<RESULT> runnable
+    )throws InvocationTargetException, InterruptedException;
```

Comment:
I think `RESULT` is too mouthful. In Java, you often see `T` because of conventions. Longer names may be used in case of ambiguity, but there's none.

Additionally, are you sure that this method should be marked as `@NotNull`? I believe the `runnable` can result in `null`, too. At least I was assuming so whilst scrolling through the code of it.

```suggestion
    <T> T executeWithProgressBlocking(
        @NotNull String operationDescription,
        @NotNull RunnableWithResult<T> runnable
    ) throws InvocationTargetException, InterruptedException;
```